### PR TITLE
Newsfeed

### PR DESCRIPTION
**Pull request by LizzieHard**
Like functionality added for posts on newsfeed

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# app/controllers/users_controller.rb
```
Comment (neoeno) [inline app/controllers/users_controller.rb]: I'm sure we can infer this information without the comment!

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,15 +1,26 @@
+<div class="container">
 
+  <div class="centered">
```
Comment (neoeno) [inline app/views/posts/index.html.erb]: Indentation is off in this file.

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,12 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
-  resources :posts
+  resources :posts do
+    member do
+      put "like" => "posts#upvote"
+      put "unlike" => "posts#downvote"
+    end
+  end
```
Comment (neoeno) [inline config/routes.rb:9]: Nice

File: coverage/.last_run.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
+  "result": {
+    "covered_percent": 100.0
+  }
+}
```
Comment (neoeno) [inline coverage/.last_run.json]: We should omit these deets (and probably `.gitignore` the coverage directory)

File: db/migrate/20170528155838_create_users.rb [hunk 5]
```diff
@@ -1,4 +1,6 @@
 class CreateUsers < ActiveRecord::Migration[5.1]
+  acts_as_voter
```
Comment (neoeno) [inline db/migrate/20170528155838_create_users.rb:2]: Does this belong in here? (I don't know, but it feels weird to have it in a migration)

File: spec/features/user_can_submit_posts_spec.rb [hunk 6]
```diff
@@ -17,11 +19,26 @@
     expect(page).to have_content("#{@time1}")
   end
 
+  it "Posts have 'like' and 'Unlike' buttons" do
+    expect(page).to have_selector(:button, 'Like')
+    expect(page).to have_selector(:button, 'Unlike')
+  end
+
+  scenario "Liking posts increases the like counter" do
+    puts Post.last.id
```
Comment (neoeno) [inline spec/features/user_can_submit_posts_spec.rb]: There's a puts still in here

File: spec/features/user_can_submit_posts_spec.rb [hunk 7]
```diff
@@ -17,11 +19,26 @@
...
+    click_link("Like")
+    expect(page).to have_content ("1 like this")
+
+  end
```
Comment (neoeno) [inline spec/features/user_can_submit_posts_spec.rb]: Nice test!

File: spec/features/user_can_submit_posts_spec.rb [hunk 8]
```diff
@@ -17,11 +19,26 @@
...
   scenario "Posts are in reverse order" do
     add_message('Message 2')
     time2 = format_time_and_date(Time.now.utc)
-    message1_string = "Message 1, created at #{@time1}"
-    message2_string = "Message 2, created at #{time2}"
-    expect(page).to have_content("#{message2_string} #{message1_string}")
+    like_message = "0 like this Like Unlike"
+    message1_string = "Message 1 #{@time1}"
+    message2_string = "Message 2 #{time2}"
+    expect(page).to have_content("#{message2_string} #{like_message} #{message1_string} #{like_message} ")
```
Comment (neoeno) [inline spec/features/user_can_submit_posts_spec.rb:38]: Good breaking out of the message

**Review by neoeno — DISMISSED** [review]

**Review by stephengeller — APPROVED** [review]

**Pull request merged**